sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuColumnarBatchSerializer.scala

```diff
@@ -245,3 +216,38 @@ private class GpuColumnarBatchSerializerInstance(
   override def deserialize[T: ClassTag](bytes: ByteBuffer, loader: ClassLoader): T =
     throw new UnsupportedOperationException
 }
+
+/**
+ * A special `ColumnVector` that describes a serialized table read from shuffle.
+ * This appears in a `ColumnarBatch` to pass serialized tables to [[ShuffleCoalesceExec]]
+ * which should always appear in the query plan immediately after a shuffle.
+ */
+class SerializedTableColumn(
+    val header: SerializedTableHeader,
+    val hostBuffer: HostMemoryBuffer) extends GpuColumnVectorBase(NullType) {
+  override def close(): Unit = {
+    if (hostBuffer != null) {
+      hostBuffer.close()
+    }
+  }
+
+  override def hasNull: Boolean = throw new IllegalStateException("should not be called")
```

Comment:
minor nit, these could be `UnsupportedOperationException`.

Reply:
I used `IllegalStateException` because calling this means the code got into an illegal state.  It should never be called.  We technically _could_ support it, but we don't ever want to.

Reply:
Ok that makes sense.